pandas/core/algorithms.py

```diff
@@ -2277,3 +2277,29 @@ def _sort_tuples(values: np.ndarray):
     arrays, _ = to_arrays(values, None)
     indexer = lexsort_indexer(arrays, orders=True)
     return values[indexer]
+
+
+def union_with_duplicates(lvals: np.ndarray, rvals: np.ndarray) -> np.ndarray:
+    """
+    Extracts the union from lvals and rvals with respect to duplicates and nans in
+    both arrays.
+
+    Parameters
+    ----------
+    lvals:
```

Comment:
types here?

Reply:
Are we duplicating them if signature is typed?

Reply:
afaik yes

Reply:
Added